app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt

```diff
@@ -227,4 +247,26 @@ class OIDCActivity: AppCompatActivity(), AccountClickListener {
         i.putExtra(ADD_NEW_ACCOUNT, true)
         startActivityForResult(i, REQUEST_NEW_ACCOUNT)
     }
+
+    override fun onPageLoaded(authorizationCode: String?, accountName: String) {
+        appCoroutineScope.launch {
+            requestAccessToken(createAuthorizationCodeFlow(accountName, serverUrl), authorizationCode)
+        }
+    }
+}
+
+interface WebViewPageLoadedListener {
+    fun onPageLoaded(authorizationCode: String?, accountName: String)
+}
+
+class OIDCWebViewClient(private val webViewPageLoadedListener: WebViewPageLoadedListener,private val accountName: String) : WebViewClient() {
```

Comment:
Make it internal and extract to separated file

Reply:
Done